SourceCgit/SourceCgit.php

```diff
@@ -249,14 +250,14 @@ public function commit_revision( $p_input ) {
 	 * @return array author / committer
 	 */
 	public function commit_author( $p_input ) {
-		preg_match( "#<tr><th>author</th><td>(.*?)<(.*?)></td><td class='right'>(.*?)</td>#", $p_input, $t_matches);
-		$t_commit['author'] = trim($t_matches[1]);
-		$t_commit['author_email'] = $t_matches[2];
-		$t_commit['date'] = date( 'Y-m-d H:i:s', strtotime( $t_matches[3] ) );
-
-		if( preg_match( "#<tr><th>committer</th><td>(.*?)<(.*?)></td><td class='right'>(.*?)</td>#", $p_input, $t_matches) ) {
-			$t_commit['committer'] = trim($t_matches[1]);
-			$t_commit['committer_email'] = $t_matches[2];
+		preg_match( "#<tr><th>author</th><td>(.*?</span>|.*?)(.*?) <((.*?@.*?)|)></td><td class='right'>(.*?)</td>#", $p_input, $t_matches);
+		$t_commit['author'] = trim($t_matches[2]);
+		$t_commit['author_email'] = $t_matches[3];
+		$t_commit['date'] = date( 'Y-m-d H:i:s', strtotime( $t_matches[4] ) );
+
+		if (preg_match( "#<tr><th>committer</th><td>(.*?</span>|.*?)(.*?) <((.*?@.*?)|)></td><td class='right'>(.*?)</td>#", $p_input, $t_matches)) {
```

Comment:
Since this regex is almost the same as the one to match author, I suggest to remove duplication by defining the repeated pattern in a variable.